src/components/Header/FlashbotsProtectStatus.tsx

```diff
@@ -0,0 +1,187 @@
+  display: none;
+  white-space: nowrap;
+  @media screen and (min-width: ${MEDIA_WIDTHS.upToSmall}px) {
+    display: block;
+    margin-right: 8px;
+  }
+`
+const SelectorControls = styled.div<{ interactive: boolean }>`
+  align-items: center;
+  background-color: ${({ theme }) => theme.bg1};
+  border: 2px solid ${({ theme }) => theme.bg1};
+  border-radius: 12px;
+  color: ${({ theme }) => theme.text1};
+  cursor: ${({ interactive }) => (interactive ? 'pointer' : 'auto')};
+  display: flex;
+  font-weight: 500;
+  justify-content: space-between;
+  padding: 6px 8px;
+`
+const ShieldWrapper = styled.div<{ active?: boolean }>`
+  height: 20px;
+  position: relative;
+  &::after {
+    content: '';
+    position: absolute;
+    background-color: ${({ theme, active }) => (active ? theme.green1 : theme.red2)};
+    width: 5px;
+    height: 5px;
+    border: 1px solid ${({ theme }) => theme.bg1};
+    border-radius: 3px;
+    top: -1px;
+    right: 5px;
+  }
+`
+
+const StatusText = styled.span<{ active: boolean }>`
+  color: ${({ theme, active }) => (active ? theme.green1 : theme.red2)};
+`
+const ShieldLogo = styled(Shield)`
+  margin-right: 8px;
+  @media screen and (min-width: ${MEDIA_WIDTHS.upToSmall}px) {
+    margin-right: 8px;
+  }
+`
+const SelectorWrapper = styled.div`
+  @media screen and (min-width: ${MEDIA_WIDTHS.upToSmall}px) {
+    position: relative;
+  }
+`
+const StyledChevronDown = styled(ChevronDown)`
+  width: 12px;
+`
+
+const Action = styled(ButtonSecondary)`
+  width: fit-content;
+  font-weight: 400;
+  font-size: 0.825rem;
+  padding: 4px 6px;
+  :hover {
+    cursor: pointer;
+    text-decoration: underline;
+  }
+`
+
+export default function FlashbotsProtect() {
+  const { chainId, library } = useActiveWeb3React()
+  const node = useRef<HTMLDivElement>()
+  const [isFlashRPC, detectFlashRPC] = useIsFlashbotsProtectRPC()
+  const open = useModalOpen(ApplicationModal.FLASHBOTS_PROTECT)
+  const toggle = useToggleModal(ApplicationModal.FLASHBOTS_PROTECT)
+  useOnClickOutside(node, open ? toggle : undefined)
+
+  const showSelector = Boolean(chainId === 1)
+
+  const conditionalToggle = useCallback(() => {
+    if (showSelector) {
+      toggle()
+    }
+  }, [showSelector, toggle])
+
+  if (!chainId || !library) {
+    return null
+  }
+
+  const shieldLogo = isFlashRPC ? flashbotsProtectShieldOn : flashbotsProtectShieldOff
+
+  return (
+    <SelectorWrapper ref={node as any}>
+      <SelectorControls onClick={conditionalToggle} interactive={showSelector}>
+        <ShieldWrapper active={isFlashRPC}>
+          <ShieldLogo src={shieldLogo} />
+        </ShieldWrapper>
+        <SelectorLabel>MEV Shield</SelectorLabel>
+        {showSelector && <StyledChevronDown />}
+      </SelectorControls>
+      {open && (
+        <FlyoutMenu>
+          <FlyoutHeader>
+            <Trans>
+              Sandwich Protection: <StatusText active={isFlashRPC}>{isFlashRPC ? 'ON' : 'OFF'}</StatusText>
```

Comment:
probably need to split up into two `<Trans>` blocks to properly extract 'on' and 'off'